runtime/src/chpl-file-utils.c

```diff
@@ -158,6 +158,39 @@ qioerr chpl_fs_is_link(int* ret, const char* name) {
   return 0;
 }
 
+qioerr chpl_fs_is_mount(int* ret, const char* name) {
+  qioerr err = 0;
+  struct stat nBuf, parentBuf;
+  int exitStatus = 0;
+  size_t nameLen = strlen(name);
+  char* parent = (char* ) chpl_mem_allocManyZero(nameLen + 4, sizeof(char), CHPL_RT_MD_OS_LAYER_TMP_DATA, 0, 0);
+  strncat(parent, name, nameLen + 1);
+  strncat(parent, "/..", 3);
+  // TODO: Using "/" is not necessarily portable, look into this
+
+  exitStatus = stat(name, &nBuf);
+  if (exitStatus) {
+    err = qio_mkerror_errno();
+    return err;
+  }
+  exitStatus = stat(parent, &parentBuf);
+  if (exitStatus) {
+    err = qio_mkerror_errno();
+  } else {
+    if (nBuf.st_dev != parentBuf.st_dev) {
+      *ret = 1;
+    // Check if the st_dev matches that of its parent directory.
+    // If they don't match, it is.
+    } else {
+      err = chpl_fs_samefile_string(ret, name, parent);
+      // If the parent directory is the same as the current directory, we've
```

Comment:
If we reach this point in the flow, can't err==0 and *ret never have been set? Or am I missing something?